shared/chat/conversation/edit-popup.desktop.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import React, {Component} from 'react'
+import {Input, PopupDialog, Box, Button} from '../../common-adapters'
+import {globalColors, globalStyles, globalMargins} from '../../styles'
+
+type Props = {
+  onClose: () => void,
+  onSubmit: (message: string) => void,
+  message: string,
+  rect: {
+    top: number,
+    left: number,
+    width: number,
+    height: number,
+  },
+}
+
+class EditPopup extends Component<void, Props, void> {
+  _input: any
+
+  _setRef = (r: any) => {
+    this._input = r
+  }
+
+  _onSave = () => {
+    this._input && this.props.onSubmit(this._input.getValue())
+    this.props.onClose()
+  }
+
+  render () {
+    const {onClose, message, onSubmit, rect} = this.props
+    // The setImmediate is due to some timing issues when injecting the portal using the unsafe method, TODO clean that up soon
```

Comment:
Agreed. This would probably be a good usecase for react-portal's built in click handling rather than `PopupDialog`, which is intended for centered dialogs.